great_expectations/metrics/metric.py

```diff
@@ -40,20 +46,15 @@ class MetaMetric(ModelMetaclass):
 
     _registry: dict[str, type["Metric"]] = {}
 
-    def __new__(cls, name, bases, attrs):
+    def __new__(cls, name, bases, attrs, **kwargs):
         register_cls = super().__new__(cls, name, bases, attrs)
         # Don't register the base Metric class
         if name != "Metric":
-            # ensure a single Domain mixin is defined
-            if len(bases) != ALLOWABLE_METRIC_MIXINS + 1 or not any(
-                issubclass(base_type, Domain) for base_type in bases
-            ):
-                raise MixinTypeError(name, "Domain")
-            try:
-                metric_name = attrs["name"]
-            except KeyError:
-                raise MissingAttributeError(name, "name")
-            MetaMetric._registry[metric_name] = register_cls
+            metric_name = attrs.get("name")
+            # Some subclasses of metric may not have a name
```

Comment:
I'm curious why we'd define a metric and then not register it. What's the use case?

Reply:
New subclasses like BatchMetric, ColumnMetric, etc